File: solutions/haskell/lucians-luscious-lasagna/1/src/LuciansLusciousLasagna.hs
module LuciansLusciousLasagna (elapsedTimeInMinutes, expectedMinutesInOven, preparationTimeInMinutes) where

expectedMinutesInOven = 40

preparationTimeInMinutes = (*) 2

elapsedTimeInMinutes layersAmount ovenMinutes =
  (+) (preparationTimeInMinutes layersAmount) (expectedMinutesInOven - ovenMinutes)

File: solutions/csharp/lucians-luscious-lasagna/3/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
	=> 40;
    
    public int RemainingMinutesInOven(int timeInOven)
	=> ExpectedMinutesInOven() - timeInOven;
    
    public int PreparationTimeInMinutes(int layers) 
	=> layers * 2;
    
    public int ElapsedTimeInMinutes(int layers, int timeInOven) 
        => timeInOven + PreparationTimeInMinutes(layers);
}


